libvast/src/segment.cpp

```diff
@@ -0,0 +1,201 @@
+/******************************************************************************
+ *                    _   _____   __________                                  *
+ *                   | | / / _ | / __/_  __/     Visibility                   *
+ *                   | |/ / __ |_\ \  / /          Across                     *
+ *                   |___/_/ |_/___/ /_/       Space and Time                 *
+ *                                                                            *
+ * This file is part of VAST. It is subject to the license terms in the       *
+ * LICENSE file found in the top-level directory of this distribution and at  *
+ * http://vast.io/license. No part of VAST, including this file, may be       *
+ * copied, modified, propagated, or distributed except according to the terms *
+ * contained in the LICENSE file.                                             *
+ ******************************************************************************/
+
+#include <caf/stream_deserializer.hpp>
+
+#include <caf/detail/scope_guard.hpp>
+
+#include "vast/bitmap_algorithms.hpp"
+#include "vast/const_table_slice_handle.hpp"
+#include "vast/ids.hpp"
+#include "vast/segment.hpp"
+#include "vast/si_literals.hpp"
+#include "vast/table_slice.hpp"
+
+#include "vast/detail/assert.hpp"
+#include "vast/detail/byte_swap.hpp"
+#include "vast/detail/coded_deserializer.hpp"
+#include "vast/detail/coded_serializer.hpp"
+#include "vast/detail/narrow.hpp"
+
+namespace vast {
+
+using namespace binary_byte_literals;
+
+namespace {
+
+template <class T>
+T to_little_endian(T x) {
+  return detail::swap<detail::host_endian, detail::little_endian>(x);
+}
+
+template <class T>
+T from_little_endian(T x) {
+  return detail::swap<detail::little_endian, detail::host_endian>(x);
+}
+
+segment::header make_header(chunk_ptr chunk) {
```

Comment:
I find this function name confusing. It doesn't really make a header, but copies the header from `chunk` and converts it to network order. IMO, since this function does two things it should get separated into two functions: `const segment::header* get_header(chunk_ptr)` and `segment::header to_network_order(const segment::header*)`.

Reply:
I can follow on the splitting aspect. But why have a function `to_network_order`? The header is written in little endian on disk, but network order is big endian. So `to_host_endian` or `from_little_endian` would make make more sense in my eyes.